src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,282 @@
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.register(this);
+        GameRegistry.register((new ItemBlock(this)).setRegistryName(this.getRegistryName()));
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+    }
+
+    public void addCollisionBoxToList(@NotNull IBlockState Tempstate,
+                                      @NotNull final World worldIn,
+                                      @NotNull final BlockPos pos,
+                                      @NotNull final AxisAlignedBB entityBox,
+                                      @NotNull final List<AxisAlignedBB> collidingBoxes,
+                                      @Nullable final Entity entityIn)
+    {
+        Tempstate = this.getActualState(Tempstate, worldIn, pos);
+        addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[0]);
+
+        if (Tempstate.getValue(NORTH))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.NORTH)]);
+        }
+
+        if (Tempstate.getValue(SOUTH))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.SOUTH)]);
+        }
+
+        if (Tempstate.getValue(EAST))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.EAST)]);
+        }
+
+        if (Tempstate.getValue(WEST))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.WEST)]);
+        }
+    }
+    private static int getBoundingBoxIndex(final EnumFacing SIDE)
+    {
+        return 1 << SIDE.getHorizontalIndex();
+    }
+
+    @NotNull
+    public AxisAlignedBB getBoundingBox(@NotNull IBlockState Tempstate,
+                                        @NotNull final IBlockAccess source,
+                                        @NotNull final BlockPos pos)
+    {
+        Tempstate = this.getActualState(Tempstate, source, pos);
+        return AABB_BY_INDEX[getBoundingBoxIndex(Tempstate)];
+    }
+
+    private static int getBoundingBoxIndex(final IBlockState state)
+    {
+        int i = 0;
+
+        if (state.getValue(NORTH))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.NORTH);
+        }
+
+        if (state.getValue(EAST))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.EAST);
+        }
+
+        if (state.getValue(SOUTH))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.SOUTH);
+        }
+
+        if (state.getValue(WEST))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.WEST);
+        }
+
+        return i;
+    }
+
+    /**
+     * Get the actual Block state of this Block at the given position. This applies properties not visible in the
+     * metadata, such as fence connections.
+     */
+    @NotNull
+    public IBlockState getActualState(@NotNull final IBlockState Tempstate,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)